lib/grandpa/message_handler.go

```diff
@@ -49,15 +48,10 @@ func (h *MessageHandler) handleMessage(from peer.ID, m GrandpaMessage) (network.
 
 	switch msg := m.(type) {
 	case *VoteMessage:
-		// send vote message to grandpa service
-		h.grandpa.in <- &networkVoteMessage{
-			from: from,
-			msg:  msg,
-		}
-
+		h.grandpa.handleVoteMessage(from, msg)
```

Comment:
handleVoteMessage should return errors and the parent here should wrap and return

Reply:
Yes, but this code is called by the network layer and it does not expect an error from a vote message we just log it, but messages like catchup should return an error to the net layer